### PR TITLE
docs: Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by rarkins**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+## Our priorities
+
+We prioritize the sustainability of this project, which means prioritizing maintainers and contributors, many of whom donate their time to helping others.
+The number one risk to Open Source projects is maintainer burnout, and the number one cause of maintainer burnout is impoliteness in various forms, usually from a minority of users.
+
+## Politeness
+
+It is unfortunately common in Open Source projects for a minority of users act with aggression and disrespect (e.g. "How dare you not fix this already"), but we do not accept it here.
+
+We don't expect too much in terms of overt politeness - simply don't be impolite.
```
Comment (viceice) [inline CODE_OF_CONDUCT.md:14]:
```suggestion
We don't expect too much in terms of over politeness - simply don't be impolite.
```

Comment (rarkins) [reply CODE_OF_CONDUCT.md:14]: Overt was intentional

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+People who work for large and/or important companies sometimes let the "pressure" of their role or task spill over into pushiness in Open Source.
+Such people sometimes find it hard to accept that maintainers cannot spend significant time helping them troubleshoot a config issue, or talk them through logs, or providing a feature or fix they need immediately.
+They are especially prone to "pinging" a maintainer with comments like `@rarkins how can we move this forward?`.
+
+Please understand that you are not our customer, and this Open Source project like any other does not owe you even the responsiveness or support you're probably already receiving.
+If you do happen to be a Mend.io customer then escalation of feature or support requests should be done via your Support or Customer Success representative and not in Open Source.
+
+## Getting more help
+
+If you need more help than this Open Source project can provide, or simply need guaranteed help, then there are two main options:
```
Comment (secustor) [inline CODE_OF_CONDUCT.md:70]:
```suggestion
If you need more help for you than this Open Source project can provide, simply need guaranteed help or need a feature which has not been prioritised high enough for you, then there are two main options:
```

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+1. Become a customer of Mend.io, such as purchasing Renovate Enterprise, or
+2. Engage privately with an experienced Renovate contributor on an hourly or project basis. Mend.io staff do not currently offer this, but one of Renovate's non-Mend.io employed maintainers who otherwise works voluntarily on the project is available for such consulting work.
```
Comment (secustor) [inline CODE_OF_CONDUCT.md:73]:
```suggestion
2. Engage privately with an experienced Renovate contributor on an hourly or project basis. Mend.io staff do not currently offer this, but one of Renovate's non-Mend.io employed maintainers ( @secustor ) who otherwise works voluntarily on the project is available for such consulting work.
```

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+When it comes to Renovate's core contributors and maintainers, we prioritize features and fixes:
+
+- those affecting a large number of users, or
+- regression errors (something used to work, but we broke it by accident), or
+- features needed by a Mend.io customer (Renovate's sponsor/owner), or
+- those we need personally, or just want to see done
+
+A surprising number of users find it hard to accept that the Renovate core team or other community contributors won't build every feature or fix every bug which they'd like to see.
+Participating in this repo is dependent on you finding a way to accept this reality, and not being pushy about it.
```
Comment (secustor) [inline CODE_OF_CONDUCT.md:45]: This sounds pretty aggressive to me.

**Review by viceice — DISMISSED** [review]

File: CODE_OF_CONDUCT.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+In other words: we can be quick to both block and unblock, and all that matters is that you can conduct yourself in future according to the project's rules.
+
+## How we prioritize work
+
+When it comes to Renovate's core contributors and maintainers, we prioritize features and fixes:
+
+- those affecting a large number of users, or
+- regression errors (something used to work, but we broke it by accident), or
+- features needed by a Mend.io customer (Renovate's sponsor/owner), or
```
Comment (secustor) [inline CODE_OF_CONDUCT.md:41]:
```suggestion
- features needed by a Mend.io customer (Renovate's sponsor/owner), or
- are sponsored by 3rd parties after independent validation ( PRs will not be accepted if they are not in the interest of Renovate OSS ), or
```
Formulating it like that so people don't think if they are ok with paying, we will implement anything.

**Comment by viceice** [comment]
needs an ls-lint exclude 

```
2024/04/25 06:13:45 CODE_OF_CONDUCT.md failed for rules: kebabcase
```

File: CODE_OF_CONDUCT.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+- are sponsored by 3rd parties after independent validation ( PRs will not be accepted if they are not in the interest of Renovate OSS ), or
+- those we need personally, or just want to see done
+
+A surprising number of users find it hard to accept that the Renovate core team or other community contributors won't build every feature or fix every bug which they'd like to see.
+Participating in this repo is dependent on you finding a way to accept this reality, and not being pushy about it.
```
Comment (rarkins) [inline CODE_OF_CONDUCT.md:46]:
```suggestion
When users are excited or pressured to get an implementing working then it can be hard to accept that there's nobody to add their particular feature or look into their bug, but it's important that you understand and accept this reality.
```

File: CODE_OF_CONDUCT.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
+# Code of Conduct
+
+In order to deliver great features and support to the Renovate Open Source project, we ask for high politeness and careful attention from those who want to join us.
```
Comment (rarkins) [inline CODE_OF_CONDUCT.md:3]:
```suggestion
In order to deliver great features and support to the Renovate Open Source project, we ask for high politeness and careful attention from those who want to participate.
```

File: CODE_OF_CONDUCT.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
+# Code of Conduct
+
+In order to deliver great features and support to the Renovate Open Source project, we ask for high politeness and careful attention from those who want to participate.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:3]:
```suggestion
To help us deliver great features and support the Renovate Open Source project we ask that you:

- are polite
- pay attention to details
- keep in mind that most maintainers are volunteers
- are respectful of the time and effort of the maintainers
```

Most people won't read the whole thing. So we should try to put the main points up front. I think this covers most of the big ticket items.

File: CODE_OF_CONDUCT.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+## Our priorities
+
+Our main goal is to keep this project sustainable.
+This involves supporting our maintainers and contributors, most of whom volunteer their time to help others.
+The biggest threat to Open Source projects is when maintainers get too stressed, and this stress is often caused by rudeness from a few users.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:9]:
```suggestion
We want to keep this project sustainable.
This means we support our maintainers and contributors, who spend their free time to help others.

Maintainers getting stressed is a big threat to Open Source projects, like ours.
Stressed maintainers quit, or reduce their time spent on the project.
Often a few users behave badly, where most users are nice.

We want to avoid maintainers getting stressed out by bad behavior from contributors.
That's why we have these rules.
```

File: CODE_OF_CONDUCT.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+
+## Politeness
+
+Sadly, it's not unusual in Open Source projects for a few users to be aggressive and disrespectful (like saying, "You should have fixed this already"), but we do not tolerate this behavior here.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:13]:
```suggestion
Sadly, it's common in Open Source projects for a few users to behave in a aggressive and rude way.
A user might say something like: "You should have fixed this bug already!", or "Why am I still waiting for this feature?".
We do not allow this kind of behavior.
```

File: CODE_OF_CONDUCT.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+We expect basic politeness—just don’t be rude.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:15]:
```suggestion
We expect basic politeness, do not act rude.
```

File: CODE_OF_CONDUCT.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+It's fine if you ask a question and don’t thank us afterward.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:16]:
```suggestion
For example: it is okay if you ask a question and do not thank us afterwards.
```

File: CODE_OF_CONDUCT.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+However, avoid making disrespectful remarks like "Pity the documentation didn’t say that" or "thanks for nothing."
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:17]:
```suggestion
But avoid writing mean comments like: "Pity the documentation didn’t say that." or "Thanks for nothing.".
```

File: CODE_OF_CONDUCT.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+## Respect the time of those who help you
+
+Respect is a two-way street, but time is limited, and those asking for help should remember that our maintainers’ time is precious.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:21]:
```suggestion
Respect goes both ways, but time is limited.
When you ask for help, please remember that the maintainer's time is valuable.
```

File: CODE_OF_CONDUCT.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+We receive many queries each week and do our best to address each one.
+Often, we need you to provide detailed logs or descriptions of your issues.
+If you’re not prepared to do this, it’s unlikely you will get the help you need.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:24]:
```suggestion
We get many questions each week and do our best to answer each one.
To get the help you need, please be prepared to give detailed logs or descriptions of your issues.
If you do not want to spend the effort giving us enough information, it's likely you will not get the help you need.
```

File: CODE_OF_CONDUCT.md [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+
+Remember, most of the support provided by our team, including Mend.io staff, is unpaid.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:26]:
```suggestion
Remember, most of the support provided by our team, including the Mend.io staff, is _unpaid_.
```

File: CODE_OF_CONDUCT.md [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+
+## Blocking and unblocking
+
+We quickly tackle rudeness in the community through automated comments, temporary blocks, or permanent bans.
+If you repeatedly break the rules (like spamming maintainers with @mentions) or challenge our guidelines openly, you will be blocked.
+We generally do not argue about these decisions, but we are willing to reverse a block if a user shows they understand and respect the rules or if there was a misunderstanding.
+
+Simply put, we block and unblock swiftly, and what matters is how you follow the rules going forward.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:34]:
```suggestion
We quickly deal with rudeness in the community with:

- automated comments
- temporary blocks
- permanent bans

If you keep breaking the rules or challenge our guidelines openly, you will be blocked.
For example: if you keep spamming the maintainers with `@mentions` or challenge our rules openly, you will be blocked.

We generally do not argue about these decisions, but we are willing to reverse a block if a you show that you understand and respect the rules, or if there was a misunderstanding.

Simply put: we block and unblock swiftly, what matters is how you follow the rules going forward.
```

File: CODE_OF_CONDUCT.md [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+
+For Renovate's core contributors and maintainers, we focus on work that:
+
+- Benefits a large number of users, or
+- Fixes regressions (errors introduced by recent changes), or
+- Is required by a Mend.io customer,
+- Is sponsored by 3rd parties after independent validation, or
+- We personally need or want to implement
+
+It may be disappointing when we can’t prioritize your specific feature or bug, but understanding and accepting this is crucial.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:46]:
```suggestion
Renovate's core contributors and maintainers focus on work that:

- Helps a lot of users, or
- Fixes regressions (errors introduced by recent changes), or
- Is required by a customer of Mend.io, or
- Is sponsored by third parties after independent validation, or
- We personally need or want to implement

You may be disappointed when we focus on other work ahead of your feature or bug, but you should understand and accept this.
```

File: CODE_OF_CONDUCT.md [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+
+## Maintaining Issue and Code Quality
+
+We use GitHub Discussions to start and sort issues but don't allow non-maintainers to directly create new issues.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:50]:
```suggestion
We use GitHub Discussions to start and sort issues.
Only maintainers are allowed to create new issues.
```

File: CODE_OF_CONDUCT.md [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+If we confirm a bug or agree with a feature idea, and if it's well-documented, we will turn it into an official issue.
+This process ensures that most issues are actionable, either by us or the community.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:52]:
```suggestion
This way most issues are ready to work on, either by us or the community.
```

File: CODE_OF_CONDUCT.md [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+
+Sometimes we reject ideas if they are too specialized and would complicate the project more than they would help.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:54]:
```suggestion
We may reject ideas that are too specialized, or that would make the project too hard to maintain.
```

File: CODE_OF_CONDUCT.md [hunk 22]
```diff
@@ -0,0 +1,81 @@
...
+
+We enforce strict coding standards and reviews to keep our codebase maintainable.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:56]:
```suggestion
We have strict coding standards and reviews to keep our code in good shape.
```

File: CODE_OF_CONDUCT.md [hunk 23]
```diff
@@ -0,0 +1,81 @@
...
+A feature or fix must not only work, it must be well designed to stay maintainable.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:57]:
```suggestion
A feature or fix must of course work, but it must also be well designed to stay maintainable.
```

File: CODE_OF_CONDUCT.md [hunk 24]
```diff
@@ -0,0 +1,81 @@
...
+We may ask for several revisions, which can be demanding, but our goal is to keep the project sustainable.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:58]:
```suggestion
We may ask you to improve your code several times in a row, which can be difficult for you.
We only do this to keep the project sustainable.
```

File: CODE_OF_CONDUCT.md [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+
+## If you have urgent work
+
+People working for significant companies might push too hard in Open Source projects.
+It’s often hard for them to understand that our maintainers cannot dedicate much time to solve their specific issues quickly.
+Frequent requests for updates like "@rarkins how can we move this forward?" are not helpful.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:64]:
```suggestion
People working for big companies might push too hard in Open Source projects.
It’s often hard for them to understand that our maintainers cannot spend much time to solve their issues quickly.
Frequent requests for updates like "@rarkins how can we move this forward?" are _not_ helpful.
```

File: CODE_OF_CONDUCT.md [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+
+Please remember, unless you are a Mend.io customer, this project does not owe you the level of response or support you might expect.
+Mend.io customers should use their designated support channels for urgent needs.
+
+## Getting more help
+
+If you need more assistance than what this project offers, there are two options:
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:71]:
```suggestion
If you need more assistance than what this project offers, you have two options:
```

File: CODE_OF_CONDUCT.md [hunk 27]
```diff
@@ -0,0 +1,81 @@
...
+
+1. Become a Mend.io customer, such as by purchasing Renovate Enterprise, or
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:73]:
```suggestion
1. Become a Mend.io customer, such as by buying Renovate Enterprise, or
```

File: CODE_OF_CONDUCT.md [hunk 28]
```diff
@@ -0,0 +1,81 @@
...
+2. Hire an experienced Renovate contributor privately for consulting. Mend.io staff do not offer this service, but one of our volunteer maintainers, `@secustor`, does.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:74]:
```suggestion
1. Hire an experienced Renovate contributor privately for consulting. Mend.io staff do _not_ offer this service, but one of our volunteer maintainers, `@secustor`, does
```

File: CODE_OF_CONDUCT.md [hunk 29]
```diff
@@ -0,0 +1,81 @@
...
+
+## Feedback
+
+We welcome respectful discussions about this policy and accept suggestions for improving this text.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:78]:
```suggestion
We welcome respectful discussions about this policy and accept suggestions that improve this text.
```

File: CODE_OF_CONDUCT.md [hunk 30]
```diff
@@ -0,0 +1,81 @@
...
+We avoid debates on social media or going off-topic in GitHub Discussions.
+
+Our consistent delivery of new features and excellent community support show that our approach is effective.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:81]:
```suggestion
Because we enforce all these rules, we can deliver new features and give excellent support to the community.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by secustor — DISMISSED** [review]

File: CODE_OF_CONDUCT.md [hunk 31]
```diff
@@ -0,0 +1,104 @@
...
+
+- Helps a lot of users, or
+- Fixes regressions (errors introduced by recent changes), or
+- Is required by a customer of Mend.io, or
+- Is sponsored by third parties after independent validation, or
+- We personally need or want to implement
+
+You may be disappointed when we focus on other work ahead of your feature or bug, but you should understand and accept this.
+
+## Maintaining Issue and Code Quality
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:69]:
```suggestion
## Maintaining Issue and Code quality
```

File: CODE_OF_CONDUCT.md [hunk 32]
```diff
@@ -0,0 +1,104 @@
...
+
+Please remember, unless you are a Mend.io customer, this project does not owe you the level of response or support you might expect.
+Mend.io customers should use their designated support channels for urgent needs.
+
+## Getting more help
+
+If you need more assistance than what this project offers, you have two options:
+
+1. Become a Mend.io customer, such as by buying Renovate Enterprise, or
+1. Hire an experienced Renovate contributor privately for consulting. Mend.io staff do _not_ offer this service, but one of our volunteer maintainers, `@secustor`, does
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:97]:
```suggestion
1. Hire an experienced Renovate contributor privately for consulting. Mend.io staff do _not_ offer this service, but one of our volunteer maintainers, but [`@secustor`](https://github.com/secustor), does
```

Might be good to link to their profile page here.

File: CODE_OF_CONDUCT.md [hunk 33]
```diff
@@ -0,0 +1,104 @@
...
+
+## Feedback
+
+We welcome respectful discussions about this policy and accept suggestions that improve this text.
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:101]:
```suggestion
We welcome respectful discussions about these rules and accept suggestions that improve this text.
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by HonkingGoose — APPROVED** [review]
We're good to go, for me.

File: CODE_OF_CONDUCT.md [hunk 34]
```diff
@@ -0,0 +1,104 @@
...
+
+Please remember, unless you are a Mend.io customer, this project does not owe you the level of response or support you might expect.
+Mend.io customers should use their designated support channels for urgent needs.
+
+## Getting more help
+
+If you need more assistance than what this project offers, you have two options:
+
+1. Become a Mend.io customer, such as by buying Renovate Enterprise, or
+1. Hire an experienced Renovate contributor privately for consulting. Mend.io staff do _not_ offer this service, but one of our volunteer maintainers, but [`@secustor`](https://github.com/secustor), does
```
Comment (viceice) [inline CODE_OF_CONDUCT.md:97]: double `but`?

Comment (HonkingGoose) [reply CODE_OF_CONDUCT.md:97]:
```suggestion
1. Hire an experienced Renovate contributor privately for consulting. Mend.io staff do _not_ offer this service, but one of our volunteer maintainers, [`@secustor`](https://github.com/secustor), does
```

You're right, good catch! :heart:

File: CODE_OF_CONDUCT.md [hunk 35]
```diff
@@ -0,0 +1,104 @@
...
+
+Respect goes both ways, but time is limited.
+When you ask for help, please remember that the maintainer's time is valuable.
+We get many questions each week and do our best to answer each one.
+To get the help you need, please be prepared to give detailed logs or descriptions of your issues.
+If you do not want to spend the effort giving us enough information, it's likely you will not get the help you need.
+
+Remember, most of the support provided by our team, including the Mend.io staff, is _unpaid_.
+
+## Blocking and unblocking
```
Comment (HonkingGoose) [inline CODE_OF_CONDUCT.md:42]: I found a section about blocking in the [`contributing.md` file on your branch](https://github.com/renovatebot/renovate/blob/docs/coc/.github/contributing.md):

> ## Rate Limiting of Support Requests through Temporary Blocking
> 
> To ensure that the Renovate maintainers don't burn out from dealing with unfriendly behavior, those who display a bad attitude when asking for or receiving support in the repo will be rate limited from further requests through the use of temporary blocking.
> The duration of the temporary block depends on how rude or inconsiderate the behavior is perceived to be, and can be from 1-30 days.
> 
> If you have been blocked temporarily and believe that it is due to a misunderstanding, or you regret your comments and wish to make amends, please reach out to the lead maintainer Rhys Arkins by email with any request for early unblocking.
> If/once you are unblocked, you should edit or delete whatever comment lead to the blocking, even if you did not intend it to be rude or inconsiderate.
> Long emails or apologies are undesirable - the maintainers are busy and want to be able to help as many users as possible with the time they have available.

I think the parts that are still relevant should be edited and moved to the Code of Conduct. Then add a link to the Code of Conduct in place of the old text.

It's probably good to mention how to get "unbanned/unblocked" in the Code of Conduct. And how long/short you want the apology to be.

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by viceice — DISMISSED** [review]

**Review by HonkingGoose — APPROVED** [review]
Good to go! 🙂

**Review by viceice — APPROVED** [review]
